### PR TITLE
Fix rubrics that hide points and have freeform comments, make sure point circles do not show

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-13727
affects: Student
release note: Fixed a bug with rubrics and free form comments

Test plan
-------------
twilson rubric course has a good amount of rubrics to use to test.  All should show correctly.  The one with no points and freeform comments should not show the point circles.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1017871&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
#### Release Note: 
Fixed a bug with rubrics and free form comments

#### Affected Apps: Student

[MBL-13727](https://instructure.atlassian.net/browse/MBL-13727)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.4% | 93.4% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Submissions/SubmissionRubric/RubricViewController.swift [hunk 1]
```diff
@@ -172,7 +172,9 @@ class RubricViewController: UIViewController {
         ratingDesc.lineBreakMode = .byWordWrapping
         ratingStack.addArrangedSubview(ratingTitle)
         ratingStack.addArrangedSubview(ratingDesc)
-        if let index = model.selectedIndex {
+        if model.freeFormCriterionComments == true && model.hideRubricPoints == true {
+            container.isHidden = true
+        } else if let index = model.selectedIndex {
```
Comment (armstrongnate) [inline Student/Student/Submissions/SubmissionRubric/RubricViewController.swift]: Mind adding a test for this? Looks like a good time to add `RubricViewControllerTests`. Not requesting we add tests for the entire view controller, just this change.

File: Student/Student/Submissions/SubmissionRubric/RubricCircleView.swift [hunk 2]
```diff
@@ -226,10 +228,20 @@ class RubricCircleView: UIView {
         let howManyCanFitInWidth = CGFloat( ceil( maxWidth / (w + space) ) )
         if howManyCanFitInWidth == 0 { return 0 }
         let rows = CGFloat(ceil(count / howManyCanFitInWidth))
+        //  If both hideRubricPoints and freeFormCriterionComments are set,
+        //  circle view should not be visible. Only comments should show of rubric.
+        if rubric.hideRubricPoints && rubric.freeFormCriterionComments { return 0 }
```
Comment (armstrongnate) [inline Student/Student/Submissions/SubmissionRubric/RubricCircleView.swift]: Looks like you check these two properties together a few different place. Do you think a getter on the rubric model would be beneficial? 

```swift
var onlyShowComments: { hideRubricPoints && freeFormCriterionComments }
```
Maybe?

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018055&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018056&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**